src/m365/aad/commands/user/user-add.ts

```diff
@@ -0,0 +1,272 @@
+          }
+        }
+
+        if (args.options.preferredLanguage && args.options.preferredLanguage.length < 2) {
+          return `'${args.options.preferredLanguage}' is not a valid preferredLanguage`;
+        }
+
+        if (args.options.firstName && args.options.firstName.length > 64) {
+          return `The maximum amount of characters for 'firstName' is 64.`;
+        }
+
+        if (args.options.lastName && args.options.lastName.length > 64) {
+          return `The maximum amount of characters for 'lastName' is 64.`;
+        }
+
+        if (args.options.jobTitle && args.options.jobTitle.length > 128) {
+          return `The maximum amount of characters for 'jobTitle' is 128.`;
+        }
+
+        if (args.options.companyName && args.options.companyName.length > 64) {
+          return `The maximum amount of characters for 'companyName' is 64.`;
+        }
+
+        if (args.options.department && args.options.department.length > 64) {
+          return `The maximum amount of characters for 'department' is 64.`;
+        }
+
+        if (args.options.managerUserName && !validation.isValidUserPrincipalName(args.options.managerUserName)) {
+          return `'${args.options.managerUserName}' is not a valid user principal name.`;
+        }
+
+        if (args.options.managerUserId && !validation.isValidGuid(args.options.managerUserId)) {
+          return `'${args.options.managerUserId}' is not a valid GUID.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['managerUserId', 'managerUserName'],
+        runsWhen: (args) => {
+          return args.options.managerId || args.options.managerUserName;
+        }
+      },
+      {
+        options: ['forceChangePasswordNextSignIn', 'forceChangePasswordNextSignInWithMfa'],
+        runsWhen: (args) => {
+          return args.options.forceChangePasswordNextSignIn || args.options.forceChangePasswordNextSignInWithMfa;
+        }
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.boolean.push('accountEnabled');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Adding user to AAD with displayName ${args.options.displayName} and userPrincipalName ${args.options.userName}`);
+    }
+
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/users`,
+        headers: {
+          'accept': 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          accountEnabled: args.options.accountEnabled || true,
+          displayName: args.options.displayName,
+          userPrincipalName: args.options.userName,
+          mailNickName: args.options.mailNickname ?? args.options.userName.split('@')[0],
+          passwordProfile: {
+            forceChangePasswordNextSignIn: args.options.forceChangePasswordNextSignIn,
```

Comment:
For my tenant, when I don't provide this value, I had to change my password. However when I explicitly set `false` in this request, I don't have to reset my password. So I guess we should make this a boolean value instead of a flag.